### PR TITLE
Chatcommand: Show help message if func returns false without message

### PR DESCRIPTION
**Pull request by HybridDog**
This makes it easier to show how to use a chatcommand to the player if he/she uses it incorrectly.
Usually, `/help <command>` should show how to invoke `<command>`, so there's often no need for individual parsing error messages.

### CODE REVIEW

**Comment by SmallJoker** [comment]
The idea is good, but the implementation looks hacky. I'd prefer to call the help function if `result == nil and success == false`, although this might cause side-effects in a few mods.

**Comment by sorcerykid** [comment]
I concur with SmallJoker, as that would be a more logical implementation.

**Comment by appgurueu** [comment]
A suggestion: Instead of modifiying on_chat_message, just modify register_chatcommand like this:
```lua
...
func = function(sendername, params)
       local success, response = def.func(sendername, params)
       if response == nil and success == false then
           success, response = core.registered_chatcommands["help"].func(name, cmd)
           return success, response
       end
end
...
```

This would also be compatible with cmdlib.

File: builtin/game/chat.lua [hunk 1]
```diff
@@ -66,8 +66,15 @@ core.register_on_chat_message(function(name, message)
 	local has_privs, missing_privs = core.check_player_privs(name, cmd_def.privs)
 	if has_privs then
 		core.set_last_run_mod(cmd_def.mod_origin)
-		local _, result = cmd_def.func(name, param)
-		if result then
+		local success, result = cmd_def.func(name, param)
+		if success == false and result == nil then
+			core.chat_send_player(name, "-!- Invalid command usage")
+			local help_def = core.registered_chatcommands["help"]
```
Comment (SmallJoker) [inline builtin/game/chat.lua:72]: Servers might remove the help command and implement something on their own. I think a `nil` check for `help_def` would be appropriate here.

Comment (HybridDog) [reply builtin/game/chat.lua:72]: I've added a check.

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**